### ISSUE TITLE
WebpackError: ReferenceError: window is not defined

### ISSUE THREAD

**Issue by danvernon** [alvarotrigo/react-fullpage#68]
Getting this error when trying to build gatsby, anyone had similar?

> 17 | })(window, function(__WEBPACK_EXTERNAL_MODULE__0__) {
     |  ^
  18 | return /******/ (function(modules) { // webpackBootstrap
  19 | /******/ 	// The module cache
  20 | /******/ 	var installedModules = {};`

WebpackError: ReferenceError: window is not defined

- react-fullpage.js:17 Object../node_modules/@fullpage/react-fullpage/dist/react-fullpage.js
    [lib]/[@fullpage]/react-fullpage/dist/react-fullpage.js:17:2

**Comment by plavreshin** [comment]
Hi @danvernon , it looks like that @cmswalker fixed this issue in `dev` branch https://github.com/alvarotrigo/react-fullpage/pull/56/commits/585c69049e755b89432ca745002737f0738853ad and it is successfully merged to master as well but yet no new version released, latest one was `0.1.`2 on 12.10 - https://github.com/alvarotrigo/react-fullpage/releases

@alvarotrigo @cmswalker  perhaps you could shed some light when to expect new release with SSR fix :) ?

**Comment by alvarotrigo** [comment]
@plavreshin thanks for the reminder!
I've released now 0.1.3 that should solve this issue.

Expect another release by the end of the month to include an updated version of fullpage.js.

**Comment by TylerBarnes** [comment]
Hey Alvaro 0.1.3 doesn't fix this. I've upgraded and I'm still getting this error:

```bash
15 |  else
  16 |          root["ReactFullpage"] = factory(root["react"]);
> 17 | })(window, function(__WEBPACK_EXTERNAL_MODULE__0__) {
     |  ^
  18 | return /******/ (function(modules) { // webpackBootstrap
  19 | /******/         // The module cache
  20 | /******/         var installedModules = {};


  WebpackError: ReferenceError: window is not defined

  - react-fullpage.js:17 Object../node_modules/@fullpage/react-fullpage/dist/react-fullpage.js
    [lib]/[@fullpage]/react-fullpage/dist/react-fullpage.js:17:2

  - bootstrap:19 __webpack_require__
    lib/webpack/bootstrap:19:1


  - bootstrap:19 __webpack_require__
    lib/webpack/bootstrap:19:1

  - sync-requires.js:15 Object../.cache/sync-requires.js
    lib/.cache/sync-requires.js:15:73

  - bootstrap:19 __webpack_require__
    lib/webpack/bootstrap:19:1

  - static-entry.js:9 Module../.cache/static-entry.js
    lib/.cache/static-entry.js:9:22

  - bootstrap:19 __webpack_require__
    lib/webpack/bootstrap:19:1

  - bootstrap:83
    lib/webpack/bootstrap:83:1


  - universalModuleDefinition:3 webpackUniversalModuleDefinition
    lib/webpack/universalModuleDefinition:3:1

  - universalModuleDefinition:10 Object.<anonymous>
    lib/webpack/universalModuleDefinition:10:2


  - static-entry.js:11 Object.Module._extensions..js
    lib/.cache/static-entry.js:11:18

  - public-page-renderer.js:8 Module.load
    lib/.cache/public-page-renderer.js:8:20

  - gatsby-browser-entry.js:36 tryModuleLoad
    lib/.cache/gatsby-browser-entry.js:36:13
```

I also tried deleting my node modules, yarn lock, and cache and public directories and reinstalling 0.1.3 but I get the error still.

**Comment by plavreshin** [comment]
@alvarotrigo  - I can confirm as well that this issue is still present using Gatsby.js but so far I have managed to use workaround provided by @TylerBarnes  and just added additionally conditional rendering into component, which relies on `ReactFullpage`.  

```
{typeof window !== 'undefined' && (
        <ReactFullpage
          licenseKey="OPEN-SOURCE-GPLV3-LICENSE"
          render={({ state, fullpageApi }) => (
            <ReactFullpage.Wrapper>{children}</ReactFullpage.Wrapper>
          )}
        />
      )}
```

**Comment by danvernon** [comment]
@plavreshin i tried your solution, and this is what happens with my build

https://determined-bartik-36155b.netlify.com/

**Comment by xvonabur** [comment]
Tried with Next.js and faced same problem. Temporary solved it by using old import:

`import ReactFullpage from '@fullpage/react-fullpage/dist/react-fullpage-commonjs';`

**Comment by danvernon** [comment]
@alvarotrigo after playing around and look at the outputted html i can see that the plugin is not setting the divs correctly 

![screenshot 2018-11-20 at 14 41 21](https://user-images.githubusercontent.com/428197/48780674-5ea59f80-ecd2-11e8-866b-489f0f044df5.png)

**Comment by cmswalker** [comment]
Sorry for the confusion over the SSR implementations. You can require the standard library now as it defaults to common JS. I have added some example repos to help

See: https://github.com/alvarotrigo/react-fullpage#server-side-rendering

**Comment by HektorCyC** [comment]
I can confirm the issue is still present in the version "^0.1.14", even using the example implementation. https://github.com/cmswalker/react-fullpage-next-example/blob/master/pages/index.js, the error is present when you try to access the url direct the error pop, funny thing it doest not pop when navigating inside the site. <Link>
`ReferenceError: window is not defined
(anonymous function)
C:/Users/EliteBook/Documents/*****/node_modules/fullpage.js/vendors/scrolloverflow.js:2132:4
Module._compile
internal/modules/cjs/loader.js:689:30
Module._extensions..js
internal/modules/cjs/loader.js:700:10
Module.load
internal/modules/cjs/loader.js:599:32
tryModuleLoad
internal/modules/cjs/loader.js:538:12
Function.Module._load
internal/modules/cjs/loader.js:530:3
Module.require
internal/modules/cjs/loader.js:637:17
require
internal/modules/cjs/helpers.js:22:18
fullpage.js/vendors/scrolloverflow
webpack:/external "fullpage.js/vendors/scrolloverflow":1
> 1 | module.exports = require("fullpage.js/vendors/scrolloverflow");
View compiled
__webpack_require__
./webpack/bootstrap:21
  18 | // Execute the module function
  19 | var threw = true;`

Im using NextJS 1.0.1.

**Comment by alvarotrigo** [comment]
@HektorCyC it seems your issue is related with scrollOverflow.
Can you please try replacing your scrollOverflow file for the latest version available in the `dev` branch of fullPage.js repository?
https://github.com/alvarotrigo/fullPage.js/blob/dev/vendors/scrolloverflow.js

Let me know if you still having the issue.

**Comment by HektorCyC** [comment]
> @HektorCyC it seems your issue is related with scrollOverflow.
> Can you please try replacing your scrollOverflow file with the latest version available in the `dev` branch of fullPage.js repository?
> https://github.com/alvarotrigo/fullPage.js/blob/dev/vendors/scrolloverflow.js
> 
> Let me know if you still having the issue.

Issue is still present, and yes the issue is with the ScrollOverflow.

**Comment by alvarotrigo** [comment]
@cmswalker any idea what would be the way to go foor this?
It seems the issue is in the last line of this portion:

```js
if ( typeof module != 'undefined' && module.exports ) {
    module.exports = IScroll;
} else if ( typeof define == 'function' && define.amd ) {
    define( function () { return IScroll; } );

    //making sure scrollOverflow works when using Require.js
    //in the browser
    if(typeof window !== 'undefined'){
        window.IScroll = IScroll;
    }
} else {
    window.IScroll = IScroll;
}

})(window, document, Math);
``` 

I guess we have to check if window exist otherwise do nothing?
But I guess there is a more elegant solution for this?

**Comment by HektorCyC** [comment]
Any update on this? @alvarotrigo @cmswalker 

:)

**Comment by alvarotrigo** [comment]
@HektorCyC I can confirm the [next.js example](https://github.com/cmswalker/react-fullpage-next-example) is not working with scrollOverflow. Didn't manage to solve it yet.

Regarding Gatsby, which is the topic of this issue, I can not reproduce the issue with it by using the [provided example](https://github.com/cmswalker/gatsby_react-fullPage_Invariant-Violation-130/tree/gatsby-fix) after changing [the import for the right file name](https://github.com/cmswalker/gatsby_react-fullPage_Invariant-Violation-130/pull/1/commits/8189300def60cbc7b36cf844166b4a0df3e3b76a).

So basically having:

```js
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import "fullpage.js/vendors/scrolloverflow"; // Optional. When using scrollOverflow:true

import ReactFullpage from "@fullpage/react-fullpage/dist/react-fullpage-umd";
```

**Comment by hskang9** [comment]
Is there any way to solve this with `componentDidMount()`?

**Comment by hskang9** [comment]
the current react wrapper just does not work when building with current gatsby version.

**Comment by wiadev** [comment]
@alvarotrigo I am still getting this error when I use `scrolloverflow` in `gatsby`

```  WebpackError: ReferenceError: window is not defined
  
  - scrolloverflow.js:2138 Object../node_modules/fullpage.js/vendors/scrolloverflow.js [as js]
    node_modules/fullpage.js/vendors/scrolloverflow.js:2138:2
```

This is what I tried

```javascript
import React from 'react'
import PropTypes from 'prop-types'
import { graphql } from 'gatsby'
import 'fullpage.js/vendors/scrolloverflow'
import ReactFullpage from '@fullpage/react-fullpage/dist/react-fullpage-umd'

export class IndexPageTemplate extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {

    if (typeof window !== 'undefined') {
      return (
        <ReactFullpage
          scrollingSpeed = {1000}
          sectionsColor={['#000000', '#000000', '#000000', '#000000', '#000000', '#000000']}
          scrollOverflow={true}
          render={({state, fullpageApi}) => {
            return (
              <ReactFullpage.Wrapper>
                <HomeSection />
                <TechnologySection />
              </ReactFullpage.Wrapper>
            )
          }}
        />
      )
    }

    return null
  }
}
```

**Comment by alvarotrigo** [comment]
@wiadev did you check the [Gatsby example provided in the project](https://github.com/alvarotrigo/react-fullpage/tree/master/examples/gatsby)?

Scrolloverflow should be working there :)


> Is there any way to solve this with componentDidMount()?

@hskang9 the issue has been closed therefore it means it has been solved  in the latest release :)

**Comment by wiadev** [comment]
@alvarotrigo I tried Gatsby example you mentioned, but I am getting this error

![Screen Shot 2019-08-19 at 04 39 30](https://user-images.githubusercontent.com/18234060/63262674-6ed7b480-c23b-11e9-9716-81fe422a124e.png)

Is this correct?

```javascript
import ReactFullpage from '@fullpage/react-fullpage/components'
import 'fullpage.js/vendors/scrolloverflow'
```

**Comment by alvarotrigo** [comment]
> import 'fullpage.js/vendors/scrolloverflow'

You won't need to import it, react-fullpage imports it for you here:
https://github.com/alvarotrigo/react-fullpage/blob/master/components/ReactFullpage/index.js#L116

**Comment by wiadev** [comment]
@alvarotrigo I removed `fullpage.js/vendors/scrolloverflow`

But I am still getting this error. `Converting circular structure to JSON`

![Screen Shot 2019-08-19 at 09 17 51](https://user-images.githubusercontent.com/18234060/63281759-43b48b80-c262-11e9-9dfc-152e8d0a33ec.png)

**Comment by wiadev** [comment]
@alvarotrigo I think this causes `circular structure to JSON` error - https://github.com/alvarotrigo/react-fullpage/blob/master/components/ReactFullpage/index.js#L243

**Comment by alvarotrigo** [comment]
@wiadev I don't think this error is related with the topic of this issue. 
Can you please open a new issue and provide an isolated reproduction of the issue if possible?
